drivers/sensor/ti/tmp435/tmp435.c

```diff
@@ -0,0 +1,216 @@
+				   uint32_t size)
+{
+	return i2c_burst_write_dt(&cfg->i2c, reg, buf, size);
+}
+
+void tmp435_get_status(const struct tmp435_config *cfg, uint8_t *status)
+{
+	int ret;
+
+	ret = tmp435_reg_read(cfg, TMP435_STATUS_REG, status, 1);
+}
+
+static int tmp435_one_shot(const struct device *dev)
+{
+	uint8_t data = 0;
+	uint8_t status = 0;
+	int ret = 0;
+	const struct tmp435_config *cfg = dev->config;
+
+	data = 1; /* write anything to start */
+	ret = tmp435_reg_write(cfg, TMP435_ONE_SHOT_START_REG, &data, 1);
+	for (uint16_t i = 0; i < TMP435_CONV_LOOP_LIMIT; i++) {
+		tmp435_get_status(cfg, &status);
+		if (status & TMP435_STATUS_REG_BUSY) {
+			/* conversion not ready */
+			k_msleep(10);
+		} else {
+			LOG_DBG("conv over, loops:%d status:%x", i, status);
+			break;
+		}
+	}
+	return ret;
+}
+
+static int tmp435_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	int ret = 0;
+	uint8_t value = 0;
+	int32_t temp = 0;
+	const struct tmp435_config *cfg = dev->config;
+	struct tmp435_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP &&
+	    chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+	tmp435_one_shot(dev); /* start conversion */
+	if ((chan == SENSOR_CHAN_ALL) || (chan == SENSOR_CHAN_DIE_TEMP)) {
+		ret = tmp435_reg_read(cfg, TMP435_LOCAL_TEMP_H_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_LOCAL_TEMP_H_REG, ret:%d", ret);
+			return ret;
+		}
+		temp = value;
+		ret = tmp435_reg_read(cfg, TMP435_LOCAL_TEMP_L_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_LOCAL_TEMP_L_REG, ret:%d", ret);
+			return ret;
+		}
+		if (value > TMP435_FRACTION_INC) {
+			temp++;
+		}
+		data->temp_die = temp + tmp435_temp_offset;
+	}
+
+	if ((chan == SENSOR_CHAN_ALL) || (chan == SENSOR_CHAN_AMBIENT_TEMP)) {
+		if (!(cfg->external_channel)) {
+			return 0; /* not enabled, just return */
+		}
+		ret = tmp435_reg_read(cfg, TMP435_REMOTE_TEMP_H_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_REMOTE_TEMP_H_REG ret:%d", ret);
+			return ret;
+		}
+		temp = value;
+		ret = tmp435_reg_read(cfg, TMP435_REMOTE_TEMP_L_REG, &value, sizeof(value));
+		if (ret < 0) {
+			LOG_ERR("Failed to read TMP435_REMOTE_TEMP_L_REG, ret:%d", ret);
+			return ret;
+		}
+		if (value > TMP435_FRACTION_INC) {
+			temp++;
+		}
+		data->temp_ambient = temp + tmp435_temp_offset;
+	}
+	return 0;
+}
+
+static int tmp435_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	int ret = 0;
+	struct tmp435_data *data = dev->data;
+	const struct tmp435_config *cfg = dev->config;
+
+	switch (chan) {
+
+	case SENSOR_CHAN_DIE_TEMP:
+		val->val1 = data->temp_die;
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		if (cfg->external_channel) {
+			val->val1 = data->temp_ambient;
+			val->val2 = 0;
```

Comment:
I didn't catch this in my previous review, but is there a conversion to standard units needed or is the raw sensor data already in degrees Celsius?

Reply:
TMP435 provides Celsius degrees + fraction, these are calculated to 1 Celsius accuracy in driver. So interfaces SENSOR_CHAN_DIE_TEMP and SENSOR_CHAN_AMBIENT_TEMP are giving temperature in degrees Celsius.